### PR TITLE
Solution for building qubes-os.org website locally on Fedora 38/39/40/41 and Debian 12

### PR DESCRIPTION
**Pull request by UndeadDevel**
The currently presented solution does not work without additional measures taken, and even then either works badly, requiring a download of different components every time the VM starts and has inconsistent results, or requires installing Ruby in addition to podman and podman-compose.

But if we're already installing Ruby to fix issues with the main solution, why not just use Ruby as the main solution? I've tested this and it works reliably, with better performance than the podman solution, without requiring podman or podman-compose, which pull far more dependencies than Ruby does (e.g. ~250 MB vs. ~30 MB additional installed size on a standard qubes Fedora template), and seems more robust.

To sum up:

- provides a working solution for users right now (podman solution does **not** work on fresh fedora-38 template)
- takes up an order of magnitude less space in the template
- has better performance
- doesn't require `sudo` when launching the server (of interest for those who use VM hardening measures)
- is actually offline; once set up does not require downloading files to run the server
- uses readily available packages from the standard repos (while all gems are installed in the AppVM)

### CODE REVIEW

**Comment by UndeadDevel** [comment]
If so desired I could also add this to the Makefile, e.g. similarly to the solution in #207

```
all:
	podman-compose up

clean:
	$(RM) -r _site

rootless:
	jekyll serve

.PHONY: all clean rootless
```
The user can then either call `sudo make` with the existing solution or `make rootless` with this solution.

Let me know if that is preferred.

**Review by marmarek — APPROVED** [review]

**Comment by marmarek** [comment]
Adding an entry to the makefile is a good idea, maybe even for the setup step too, like `rootless-setup` target?

**Comment by UndeadDevel** [comment]
> Adding an entry to the makefile is a good idea, maybe even for the setup step too, like `rootless-setup` target?

Done. Tested on updated Fedora-38-based AppVM and StandaloneVM.

**Comment by UndeadDevel** [comment]
@marmarek I've now tested this on fresh templates of `fedora-38`, `fedora-38-xfce`, `debian-12` and `debian-12-xfce` on Qubes 4.2rc5 and updated the instructions accordingly (they didn't change for fedora, but debian needed some adjustment). I think it's good to go.

**Comment by UndeadDevel** [comment]
PipelineRefresh

**Comment by UndeadDevel** [comment]
PipelineRetry

**Comment by marmarek** [comment]
PipelineRetry

**Comment by marmarek** [comment]
> Hmm...doesn't work

There is an access list who can issue commands...

**Comment by UndeadDevel** [comment]
> There is an access list who can issue commands...

I suspected...well the failure is the same as in all the other doc PRs (not due to this PR):

```
For the Links > Internal check, the following failures were found:
* At ./_site/doc/device-handling-security/index.html:424:
  internally linking to /news/2018/09/11/qubes-ctap-proxy/, which does not exist
* At ./_site/news/2018/09/11/qubes-u2f-proxy/index.html:364:
  internally linking to /attachment/doc/u2f.svg, which does not exist
HTML-Proofer found 2 failures!
```

**Comment by unman** [comment]
On Thu, Apr 18, 2024 at 02:31:43AM -0700, UndeadDevel wrote:
> I suspected...well the failure is the same as in all the other doc PRs (not due to this PR):

Fixed both of these errors

**Comment by UndeadDevel** [comment]
> Fixed both of these errors

Nice! It might be worth running PipelineRetry on all the PRs with integration check failed status in qubes-doc as well as this one; or test on one (e.g. this one) to see if something else doesn't come up in the check and if not, then all the others with failed checks (in qubes-doc) as most are sitting there waiting for review, so there likely won't be more commits to trigger a new check.

**Comment by UndeadDevel** [comment]
Tested both `fedora-40` and `fedora-40-xfce` now and both work perfectly well with this solution; updated the README.md to reflect that.

Since this is probably now the most thoroughly tested PR in this repo and the first templates I tested (fedora-38 ones) are now EOL, thus making that effort a total waste, and no reason has been given why this is not getting merged, the latest Fedora templates will be the last I'll have tested for this PR.

**Comment by parulin** [comment]
I tried to follow the instructions of the main branch on Debian 12 and it failed, following this PR helped me.

I don't know if it's relevant (because I didn't try exactly theses rootless instructions in the first place) but I got this error:

> You have already activated jekyll-sass-converter 3.0.0, but your Gemfile requires jekyll-sass-converter 1.5.2.
> Prepending `bundle exec` to your command may solve this. (Gem::LoadError)

Replacing `jekyll serve` by `bundle exec jekyll serve` works for me.

**Review by marmarek — APPROVED** [review]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

Approved as of f0ad5c73070e5141a94bc5bb2f53e7481bb2ea9f
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEEnYWYU8Ll1fIX9iss8yiUvpaEk4oFAmZuQn8ACgkQ8yiUvpaE
k4q03xAAnBNaphhqW/5s+sBDY2NjjUrvWV1D8lgn4d7p92xp/70U2GSRlVkTx6z6
peuO4DOtcNV2FVdE1ovOjLcLCoJXaph4qtEGoqS7dvpcajjalW47zVscD78bwc0u
1cej1I0IOl/puS+MzNBxc8BYb7Rhe+LUELx0J8mtlsBEZ1pCbT5MBPwvPCIX1Hr1
m1IzkXH7EULQS/fy6JpqZp5A9nZDWjBYxm2WPN9drKCx64UJaxWMFMBkskCcfVGr
wFO7URJi8KazD9y9RlWCeQJuL07GEbFvv5KOWAn92xt5eXLJYxoUANPKkUfa0NYO
wUxeP/OApWJVaik1b5WTtX8pHi16yNBdFPqWwUVVNPaEI0Ts/MikdPt3Llj94XUe
XR319j9Ce48n7TwPeOr45a6f0ZfCYm48OfHrJR3BV/N7tVkdu2PbbFNMZfTwBleY
NbQvlwXvWppnaVVgUsMb4E/CHWtZwGldWRtbSeqm0n6aByd78neTl6KJVPkGC7+t
/rZldjBcZSKDjBDZYt8hDEwRLUaeqqs8N8E7Qn0H6+5K/R7UAnYpblSSpsCqy/3L
onPVsOvSHdzvmIYaR5XH9i9Ub1hzxEW8U+qAprJEAKF+1igp5sjoAa4Ur3XtVeKa
kr8JxovJRghhAdObJdAKnlc7bVce/qI1+IYl0r+4qqG+B6M/UEs=
=LPtk
-----END PGP SIGNATURE-----
```

**Comment by UndeadDevel** [comment]
The same instructions still work for `fedora-41-xfce`...time to merge?
